android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewHolder.java

```diff
@@ -412,8 +412,15 @@ private void setHeaderFooter(TiViewProxy listViewProxy,
 
 						ConstraintSet constraintSet = new ConstraintSet();
 						constraintSet.clone((ConstraintLayout) this.container);
-						constraintSet.connect(R.id.titanium_ui_listview_holder_content, ConstraintSet.BOTTOM,
-							R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
+						if (this.content.getChildCount() > 0) {
+							// attach to content
+							constraintSet.connect(R.id.titanium_ui_listview_holder_content, ConstraintSet.BOTTOM,
+								R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
+						} else {
+							// attach to header
+							constraintSet.connect(R.id.titanium_ui_listview_holder_header, ConstraintSet.BOTTOM,
+								R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
+						}
```

Comment:
Generally when you have a duplicate function call with many shared arguments, it's easier to read and maintain if you set the thing that changed in its own variable like this:

```suggestion

						int holderId = R.id.titanium_ui_listview_holder_header;
						if (this.content.getChildCount() > 0) {
							holderId = R.id.titanium_ui_listview_holder_content;
						}
						
						constraintSet.connect(holderId, ConstraintSet.BOTTOM,
							R.id.titanium_ui_listview_holder_footer, ConstraintSet.TOP, 0);
```

Also, `holderId` is a horrible name. The argument is `startId`.

Reply:
I'll change it to `constraintStartId` but I like this way

Reply:
@cb1kenobi "holder" is a actually a common terminology in the Android world for these cases, especially with [`RecyclerView` instances](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.ViewHolder). I would keep it if it's okay for you :)